Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery+ImagePreview.swift

```diff
@@ -87,15 +87,15 @@ extension ChatMessageGalleryView {
             let attachment = content
 
             loadingIndicator.isVisible = true
-            imageTask = imageView
-                .loadImage(
-                    from: attachment?.payload.imagePreviewURL,
-                    resize: false,
-                    components: components
-                ) { [weak self] _ in
+            imageTask = components.imageLoader.loadImage(
+                into: imageView,
+                url: attachment?.payload.imagePreviewURL,
+                imageCDN: components.imageCDN,
+                completion: { [weak self] _ in
```

Comment:
Hm, is it guaranteed the completion is called on main thread? 🤔

Reply:
Yes, Nuke guarantees that the completion gets called on the main thread. In case the customer provides own loader, they will need to ensure this